package.json

```diff
@@ -77,16 +79,20 @@
     "power-assert": "1.4.2",
     "rimraf": "2.6.1",
     "semistandard": "10.0.0",
-    "swig": "1.4.2"
+    "source-map-support": "^0.4.14",
+    "swig": "1.4.2",
+    "typescript": "^2.2.2"
   },
   "scripts": {
-    "lint": "semistandard \"**/*.js\"",
-    "generate-apis": "node scripts/generate",
+    "lint": "semistandard \"samples/**/*.js\"",
+    "generate-apis": "npm run build-tools && node scripts/generate.js",
```

Comment:
Yes :) `npm run build-tools` generates `generate.js`, and `npm run generate-apis` generates `apis/*.ts`, which `npm run build` compiles into js.  Sorry - I know this is a little confusing.